pytorch_lightning/trainer/trainer.py

```diff
@@ -96,7 +96,7 @@ def __init__(
             num_processes: int = 1,
             gpus: Optional[Union[List[int], str, int]] = None,
             auto_select_gpus: bool = False,
-            num_tpu_cores: Optional[int] = None,
+            tpu_cores: Optional[Union[List[int], int]] = None,
```

Comment:
we need to add deprecation warnings for the other arg no?

Reply:
Gotcha! I'll keep the `num_tpu_cores` and add a  deprecation warning.

Reply:
if you keep `num_tpu_cores` no deprecation is needed

Reply:
@Borda My idea is to replace `num_tpu_cores` with `tpu_cores` in the coming iterations. As @williamFalcon mentioned, I plan to support `num_tpu_cores` for now with a deprecation warning. Will that be fine?

Reply:
oh, I see now, sorry for the confusion... add deprecation for `num_tpu_cores`

Reply:
cool :))